distributed/diagnostics/nvml.py

```diff
@@ -1,19 +1,31 @@
 import pynvml
 
-pynvml.nvmlInit()
-count = pynvml.nvmlDeviceGetCount()
+need_pynvml_init = True
+handles = None
 
-handles = [pynvml.nvmlDeviceGetHandleByIndex(i) for i in range(count)]
+
+def _initialize_pynvml():
+    global need_pynvml_init, handles
+    pynvml.nvmlInit()
+    count = pynvml.nvmlDeviceGetCount()
+    handles = [pynvml.nvmlDeviceGetHandleByIndex(i) for i in range(count)]
+    need_pynvml_init = False
 
 
 def real_time():
+    global need_pynvml_init, handles
```

Comment:
Can you explain why this is needed/works? It seems like the old code should handle the init on import.

Reply:
@rlratzel - Sorry I missed your code snippet in the slack thread (long thread!).

The short answer is "I dont know yet"...  However,  it may have something to do with multiple workers fighting over a "load" lock for the `nvml` library... I should probably understand pynvm well enought to answer this, so I'll dig a bit deeper :)